src/Twig/Extension/SonataAdminExtension.php

```diff
@@ -96,14 +109,14 @@ public function __construct(
         $translator = null,
```

Comment:
The translator with also be remove in next major. And the logger too I think.

So it will be 
````
__construct(Pool $pool, ?ContainerInterface $templateRegistries, PropertyAccessor $propertyAccessor)
```
And to allow people using the new signature, we have to do
```
__construct(
Pool $pool
$templateRegistriesOrLogger = null,
$propertyAccessorOrTranslator = null
$deprecatedTemplateRegistries = null
$deprecatedPropertyAccessorOrSecurityChecker = null
$securityChecker = null
) {

$this->pool = $pool;

if ($templateRegistriesOrLogger instanceof ContainerInterface) {
    $this->templateRegistries = $templateRegistriesOrLogger;

    if (!$propertyAccessorOrTranslator instanceof PropertyAccessor) {
        throw \TypeError();
    }
    
    $this->propertyAccessor = $propertyAccessorOrTranslator;
    
    return;
} elseif ($templateRegistriesOrLogger instanceof LoggerInterface || null === $templateRegistriesOrLogger) {
    // trigger a deprecation

    ...
} else {
    throw TypeError;
}
```

And
- We pass only the 3 params needed to this extension in the config
- We remove the declaration of the filter/function/... in this class (but keep the method)
- We do something similar to https://github.com/sonata-project/SonataAdminBundle/blob/3.x/src/Admin/AdminHelper.php#L141 for deprecated method if the property is not filled anymore by the constructor.

WDYT @franmomu @tambait ?

Reply:
Yes, I wanted to ask about translator. In my initial change I was thinking to deprecate/remove it but then I saw `NEXT_MAJOR: make the translator parameter required` so wasn't sure if that's still valid or I miss something :)

Reply:
It's not valid anymore, since translator is only used in `getXEditableChoices` which will be moved, isn't it ?

Reply:
True. But I see now that my IDE is, for some reason, giving me wrong signals about the errors that confused me over this.